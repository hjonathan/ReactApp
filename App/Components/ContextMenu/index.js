import React, {Component} from "react";
import {View, Text, TouchableOpacity} from "react-native";
import MaterialIcon from "react-native-vector-icons/MaterialIcons";
import I18n from "../../I18n";
import Styles from "./styles";

/**
 * Context menu class
 * Example:
 *      ...
 *      import ContextMenu from "../../Components/ContextMenu";
 *
 *      class SomeComponent extends component {
 *          render () {
 *              return (
 *                  <ContextMenu
 *                      type="audio" // audio \\ image \\ video
 *                      onClose={()=>{}} // close function
 *                      takeOne={() => {}} // creates a file function
 *                      selectFile={() => {}} // access to gallery
 *                  />
 *              );
 *          }
 *      }
 */
class ContextMenu extends Component {
    /**
     * Closes view
     */
    onCloseMenu () {
        if (this.props.onClose) {
            this.props.onClose({status: false});
        }
    }

    /**
     * Executes takeOne function
     */
    takeOne () {
        if (this.props.takeOne) {
            this.props.takeOne({status: true});
        }
    }

    /**
     * Executes selectFile function
     */
    selectFile () {
        if (this.props.selectFile) {
            this.props.selectFile();
        }
    }

    /**
     * Renders header
     * @returns {*}
     */
    renderHeader () {
        return (
            <View style={Styles.headerContainer}>
                <View style={Styles.headerTitle}>
                    <Text style={Styles.headerText}>
                        {I18n.t("selectAudioDialogtitle")}
                    </Text>
                </View>
                <TouchableOpacity
                    onPress={() => {
                        this.onCloseMenu();
                    }}
                >
                    <MaterialIcon name="clear" style={Styles.closeIcon} />
                </TouchableOpacity>
            </View>
        );
    }

    /**
     * Renders take one button
     * @returns {*}
     */
    renderTakeButton () {
        return (
            <View style={Styles.buttonContainer}>
                <TouchableOpacity
                    onPress={() => {
                        this.takeOne();
                    }}
                >
                    <View style={Styles.button}>
                        <MaterialIcon name="mic" style={Styles.buttonIcon} />
                    </View>
                </TouchableOpacity>
                <View style={Styles.labelButton}>
                    <Text style={{textAlign: "center"}}>{I18n.t("selectAudioStringDialog")}</Text>
                </View>
            </View>
        );
    }

    /**
     * Renders gallery button
     * @returns {*}
     */
    renderGalleryButton () {
        return (
            <View style={Styles.buttonContainer}>
                <TouchableOpacity
                    onPress={() => {
                        this.selectFile();
                    }}
                >
                    <View style={Styles.button}>
                        <MaterialIcon name="graphic-eq" style={Styles.buttonIcon} />
                    </View>
                </TouchableOpacity>
                <View style={Styles.labelButton}>
                    <Text style={{textAlign: "center"}}>{I18n.t("selectGalleryStringAudioDialog")}</Text>
                </View>
            </View>
        );
    }

    /**
     * Renders component
     * @returns {*}
     */
    render () {
        return (
            <View style={Styles.mainContainer}>
                <View style={Styles.subContainer}>
                    {this.renderHeader()}
                    <View style={Styles.panelContainer}>
                        {this.renderTakeButton()}
                        {this.renderGalleryButton()}
                    </View>
                </View>
            </View>
        );
    }
}

export default ContextMenu;
