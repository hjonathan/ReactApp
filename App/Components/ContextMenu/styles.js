import {StyleSheet} from "react-native";
import * as Res from "../../Assets/resources";

export default StyleSheet.create({
    mainContainer: {
        flex: 1,
        alignItems: "center",
        justifyContent: "center"
    },
    subContainer: {
        width: "70%",
        flexDirection: "column",
        justifyContent: "center"
    },
    headerContainer: {
        flexDirection: "row",
        alignItems: "center",
        justifyContent: "space-between",
        backgroundColor: Res.colors.lightColor,
        height: 38,
        padding: 8
    },
    headerTitle: {
        flex: 1,
        justifyContent: "center"
    },
    headerText: {
        textAlign: "center",
        color: Res.colors.white
    },
    closeIcon: {
        color: Res.colors.white,
        fontSize: 28
    },
    panelContainer: {
        flexDirection: "row",
        backgroundColor: Res.colors.white,
        padding: 16
    },
    buttonContainer: {
        flexWrap: "wrap",
        alignItems: "center",
        width: "50%"
    },
    button: {
        borderRadius: 8,
        padding: 4,
        backgroundColor: Res.colors.mainColor
    },
    buttonIcon: {
        color: Res.colors.white,
        fontSize: 36
    },
    labelButton: {
        flexDirection: "row",
        flexWrap: "wrap",
        justifyContent: "center"
    }
});
