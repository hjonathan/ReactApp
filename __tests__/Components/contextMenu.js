import 'react-native';
import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import ContextMenu from '../../App/Components/ContextMenu';

describe('Component >>> ContextMenu', () => {
    const onClose = () => {},
        takeOne = () => {},
        selectFile = () => {};

    let contextMenu,
        contextMenuDefault;

    beforeEach(() => {
        contextMenuDefault = shallow(<ContextMenu />);
        contextMenu = shallow(<ContextMenu onClose={onClose} takeOne={takeOne} selectFile={selectFile} />);
    });

    test('Render contextMenu', () => {
        expect(toJson(contextMenu)).toMatchSnapshot();
        expect(toJson(contextMenuDefault)).toMatchSnapshot();
    });
    test('Render renderHeader', () => {
        let header,
            close;
        header = contextMenu.instance().renderHeader();
        close = header.props.children[1].props;
        expect(close.onPress()).toMatchSnapshot();
    });

    test('Render onCloseMenu', () => {
        expect(contextMenu.instance().onCloseMenu()).toMatchSnapshot();
        expect(contextMenuDefault.instance().onCloseMenu()).toMatchSnapshot();
    });

    test('Render takeButton', () => {
        let takeButton,
            takeButtonDef;
        takeButton = contextMenu.instance().renderTakeButton();
        takeButtonDef = contextMenuDefault.instance().renderTakeButton();
        expect(takeButton.props.children[0].props.onPress()).toMatchSnapshot();
        expect(takeButtonDef.props.children[0].props.onPress()).toMatchSnapshot();
    });

    test('Render galleryButton', () => {
        let galleryButton,
            galleryButtonDef;
        galleryButton = contextMenu.instance().renderGalleryButton();
        galleryButtonDef = contextMenuDefault.instance().renderGalleryButton();
        expect(galleryButton.props.children[0].props.onPress()).toMatchSnapshot();
        expect(galleryButtonDef.props.children[0].props.onPress()).toMatchSnapshot();
    });
});
