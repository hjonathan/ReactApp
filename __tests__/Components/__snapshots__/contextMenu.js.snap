// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component >>> ContextMenu Render contextMenu 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "flexDirection": "column",
        "justifyContent": "center",
        "width": "70%",
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#3397E1",
          "flexDirection": "row",
          "height": 38,
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Component
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFF",
              "textAlign": "center",
            }
          }
        >
          selectAudioDialogtitle
        </Text>
      </Component>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Icon
          allowFontScaling={false}
          name="clear"
          size={12}
          style={
            Object {
              "color": "#FFF",
              "fontSize": 28,
            }
          }
        />
      </TouchableOpacity>
    </Component>
    <Component
      style={
        Object {
          "backgroundColor": "#FFF",
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flexWrap": "wrap",
            "width": "50%",
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Component
            style={
              Object {
                "backgroundColor": "#1F6CBC",
                "borderRadius": 8,
                "padding": 4,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              name="mic"
              size={12}
              style={
                Object {
                  "color": "#FFF",
                  "fontSize": 36,
                }
              }
            />
          </Component>
        </TouchableOpacity>
        <Component
          style={
            Object {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            selectAudioStringDialog
          </Text>
        </Component>
      </Component>
      <Component
        style={
          Object {
            "alignItems": "center",
            "flexWrap": "wrap",
            "width": "50%",
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Component
            style={
              Object {
                "backgroundColor": "#1F6CBC",
                "borderRadius": 8,
                "padding": 4,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              name="graphic-eq"
              size={12}
              style={
                Object {
                  "color": "#FFF",
                  "fontSize": 36,
                }
              }
            />
          </Component>
        </TouchableOpacity>
        <Component
          style={
            Object {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            selectGalleryStringAudioDialog
          </Text>
        </Component>
      </Component>
    </Component>
  </Component>
</Component>
`;

exports[`Component >>> ContextMenu Render contextMenu 2`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "flexDirection": "column",
        "justifyContent": "center",
        "width": "70%",
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#3397E1",
          "flexDirection": "row",
          "height": 38,
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Component
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFF",
              "textAlign": "center",
            }
          }
        >
          selectAudioDialogtitle
        </Text>
      </Component>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Icon
          allowFontScaling={false}
          name="clear"
          size={12}
          style={
            Object {
              "color": "#FFF",
              "fontSize": 28,
            }
          }
        />
      </TouchableOpacity>
    </Component>
    <Component
      style={
        Object {
          "backgroundColor": "#FFF",
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flexWrap": "wrap",
            "width": "50%",
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Component
            style={
              Object {
                "backgroundColor": "#1F6CBC",
                "borderRadius": 8,
                "padding": 4,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              name="mic"
              size={12}
              style={
                Object {
                  "color": "#FFF",
                  "fontSize": 36,
                }
              }
            />
          </Component>
        </TouchableOpacity>
        <Component
          style={
            Object {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            selectAudioStringDialog
          </Text>
        </Component>
      </Component>
      <Component
        style={
          Object {
            "alignItems": "center",
            "flexWrap": "wrap",
            "width": "50%",
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Component
            style={
              Object {
                "backgroundColor": "#1F6CBC",
                "borderRadius": 8,
                "padding": 4,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              name="graphic-eq"
              size={12}
              style={
                Object {
                  "color": "#FFF",
                  "fontSize": 36,
                }
              }
            />
          </Component>
        </TouchableOpacity>
        <Component
          style={
            Object {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            selectGalleryStringAudioDialog
          </Text>
        </Component>
      </Component>
    </Component>
  </Component>
</Component>
`;

exports[`Component >>> ContextMenu Render galleryButton 1`] = `undefined`;

exports[`Component >>> ContextMenu Render galleryButton 2`] = `undefined`;

exports[`Component >>> ContextMenu Render onCloseMenu 1`] = `undefined`;

exports[`Component >>> ContextMenu Render onCloseMenu 2`] = `undefined`;

exports[`Component >>> ContextMenu Render renderHeader 1`] = `undefined`;

exports[`Component >>> ContextMenu Render takeButton 1`] = `undefined`;

exports[`Component >>> ContextMenu Render takeButton 2`] = `undefined`;
